MetasysServices/Trends/TrendServiceProvider.cs

```diff
@@ -140,11 +140,14 @@ public async Task<PagedResult<Sample>> GetSamplesAsync(Guid objectId, AttributeE
                 string unitsUrl = string.Empty;
                 try
                 {
-                    sample.Timestamp = s["timestamp"].Value<DateTime>();
-                    sample.IsReliable = s["isReliable"].Value<Boolean>();
-                    sample.Value = s["value"]["value"].Value<double>();
-                    unitsUrl = s["value"]["units"].Value<string>();
-                    sample.Unit = ResourceManager.Localize(unitsUrl, _CultureInfo);
+                    sample.Timestamp = (DateTime)s["sampleTime"];
+                    if (s["reliability"] != null && s["reliability"].ToString() == "reliabilityEnumSet.reliable")

```

Comment:
According to the schema `reliability` is required so the null check should be unnecessary.

<img width="770" alt="image" src="https://github.com/user-attachments/assets/ad1ac610-d587-4001-9503-8c179a921e97">

Also, I'd recommend sticking to one convention of retrieving properties from a token and converting them, that is the use of `.Value<T>` method.

So

```suggestion
                    sample.Timestamp = s["sampleTime"].Value<DateTime>();
                    sample.IsReliable = s["reliability"].Value<string>() == Variant.ReliableString;
                    sample.Value = s["value"].Value<double>();
```

(In my suggestion I added an internal static property named Reliable on Variant. @Federico-Artioli  may not agree with that location but it seems we should define it somewhere.

@Federico-Artioli  Looks like `units` is not defined in the spec. If we want units we'd need to make an additional call.

Reply:
Also we should probably have some unit tests for this. Unit tests are pretty easy to write due to the use of the httpTest mock object provided by Fluent.Http.

If you don't have time to do that, then just finish up by writing up a CR and assigning it to Federico.